models/object_detection_nanodet/demo.py

```diff
@@ -0,0 +1,122 @@
+        image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+
+        srcimg = image
+        drawimg = srcimg.copy()
+        a = time.time()
+        #image = model_net.infer(image)
+        left, top, ratioh, ratiow, det_bboxes, det_conf, det_classid = model_net.infer(image)
+        b = time.time()
+        print('Inference_Time:'+str(b-a)+' secs')
+        for i in range(det_bboxes.shape[0]):
+            xmin, ymin, xmax, ymax = max(int((det_bboxes[i,0] - left) * ratiow), 0), max(int((det_bboxes[i,1] - top) * ratioh), 0), min(
+                int((det_bboxes[i,2] - left) * ratiow), srcimg.shape[1]), min(int((det_bboxes[i,3] - top) * ratioh), srcimg.shape[0])
+            frame = drawPred(drawimg, det_classid[i], det_conf[i], xmin, ymin, xmax, ymax)
+
+        frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+        cv2.namedWindow(args.image_path, cv2.WINDOW_AUTOSIZE)
+        cv2.imshow(args.image_path, frame)
+        cv2.waitKey(0)
+
+        if args.save:
+            print('Resutls saved to result.jpg\n')
+            cv2.imwrite('result.jpg', frame)
+
+    else:
+        print("Press 1 to stop video capture")
+        cap = cv2.VideoCapture(0)
+        tm = cv2.TickMeter()
+        total_frames = 0
+        frame_list = []
+        Video_save = False
+        if(args.save):
+            Video_save = True
+
+        while cv2.waitKey(1) < 0:
+            hasFrame, frame = cap.read()
+            if not hasFrame:
+                print('No frames grabbed!')
+                break
+
+            frame = cv2.flip(frame, 1)
+            srcimg = frame
+            drawimg = srcimg.copy()
+            #frame = cv2.resize(frame, [args.width, args.height])
+            # Inference
+            tm.start()
+            left, top, ratioh, ratiow, det_bboxes, det_conf, det_classid = model_net.infer(frame)
+            tm.stop()
+
+            for i in range(det_bboxes.shape[0]):
+                xmin, ymin, xmax, ymax = max(int((det_bboxes[i,0] - left) * ratiow), 0), max(int((det_bboxes[i,1] - top) * ratioh), 0), min(
+                    int((det_bboxes[i,2] - left) * ratiow), srcimg.shape[1]), min(int((det_bboxes[i,3] - top) * ratioh), srcimg.shape[0])
+                image = drawPred(drawimg, det_classid[i], det_conf[i], xmin, ymin, xmax, ymax)
+
+            total_frames += 1
+            fps=tm.getFPS()
+
+            if fps > 0:
+                fps_label = "FPS: %.2f" % fps
+                cv2.putText(image, fps_label, (10, 25), cv2.FONT_HERSHEY_SIMPLEX, 1, (0, 0, 255), 2)
+
+            cv2.imshow("output", image)
+
+            if cv2.waitKey(1) > -1:
+                print("Stream terminated")
+                break
+
+            if(args.save):
+                image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+                frame_list.append(image)
+
+        if(Video_save):
+            clip = moviepy.video.io.ImageSequenceClip.ImageSequenceClip(frame_list, fps=fps)
+            clip.write_videofile('Webcam_result.mp4')
```

Comment:
Use `VideoWriter` instead.

Reply:
- Resolved changes: removed moviepy and replaced it with opencv.Videowriter